### DISCUSSION TITLE
How to get the batch features of line graph

### DISCUSSION THREAD

**Discussion by StefanIsSmart** [pyg-team/pytorch_geometric#8763] [Q&A]
As we all know, the pyg could convert a graph to a linegraph.
But finished converting, we cannot get the new batch features of the edges. The batch is still representative the original nodes rather than the nodes that converted from edges. 

How to get the batch features of new graph?

**Comment by rusty1s** [comment]
What do you mean by batch features?

**Reply by StefanIsSmart** [reply]
My meaning is the new_batch of the converted line-graph.

**Reply by rusty1s** [reply]
Do you mean that `data.batch` is not correct anymore when running `LineGraph` on a `Batch` object created by `DataLoader`?

**Reply by StefanIsSmart** [reply]
Yes, because when I convert the graph to a line graph, the data.batch is not changed, still is the batch of the original graph.

I think the line graph needs a new batch of the line graph so that it can be updated as a normal graph.

**Reply by rusty1s** [reply]
That is correct; most transforms in PyG are designed to operate on single graphs only. The current procedure would be to convert each graph separately via the `LineGraph` transform, and then batch the line-graphs together afterwards. That is what is happening if you use, e.g., `Dataset.transform` or `Dataset.pre_transform`. Would that be applicable in your use-case?

**Reply by StefanIsSmart** [reply]
Thank you very much and I will have a try and tell you the result!

So the transformation of a batch of `node graph` to `line-graph` will be correct right?
I ask this question because you said: `most transforms in PyG are designed to operate on single graphs only.`
I worry I converting a batch of `node graph` will have some Errors. :p

Thank you again for your reply!

**Reply by rusty1s** [reply]
Yes, in general it is not guaranteed that transforms work on a batch of graphs (i.e., `batch` vector is not maintained). This may not always be a problem, but for some transformations it is. For example, `Center` would center on the center of all nodes in the batch, not just the nodes inside each graph.

**Reply by StefanIsSmart** [reply]
As your answer in another discussion.
The batch info of ling_graph could get from :
`batch[edge_index[0]]`
Right?

**Reply by rusty1s** [reply]
Yes.
